spec/lang/step/operators.md

```diff
@@ -15,22 +15,56 @@ impl<M: Memory> Machine<M> {
 
 ```rust
 impl<M: Memory> Machine<M> {
-    fn eval_int_un_op(op: IntUnOp, operand: Int) -> Result<Int> {
+    /// Perform the operation on the mathematical integer `operand`,
+    /// but correcting for non-pure effects dependent of the `operand_ty`.
+    fn eval_int_un_op(op: IntUnOp, operand: Int, operand_ty: IntType) -> Result<Int> {
         use IntUnOp::*;
         ret(match op {
-            Neg => -operand,
-            BitNot => !operand,
+            // Put the result into the right range (in case of overflow).
+            Neg => operand_ty.bring_in_bounds(-operand),
+            // Put the result into the right range (in case of a unsigned numbers).
```

Comment:
```suggestion
            // Put the result into the right range (in case of a unsigned numbers, which `!`
            // makes negative by inverting all the leading zeros).
```